core/src/main/java/org/apache/accumulo/core/client/impl/TabletServerBatchWriter.java

```diff
@@ -699,7 +701,16 @@ else if (Tables.getTableState(context.getInstance(), table) == TableState.OFFLIN
 
     }
 
-    void addMutations(MutationSet mutationsToSend) {
+    void queueMutations(final MutationSet mutationsToSend) {
+      queueThreadPool.submit(new Runnable() {
```

Comment:
I think it be simpler to use a producer/consumer model for the `MutationSet`'s with a single consumer thread instead of constantly submitting new `Runnable`'s. The consumer thread can essentially do the same thing: poll for a `MutationSet`, and then run `addMutations` with that object. It eliminates the need for another ExecutorService and repeatedly creating new Runnables. We might also be able to implement some back-pressure on the client via that Queue as well.